pkg/registry/apis/provisioning/files.go

```diff
@@ -147,13 +148,37 @@ func (s *filesConnector) Connect(ctx context.Context, name string, opts runtime.
 	}), nil
 }
 
-func (s *filesConnector) getParser(repo repository.Repository) (*resources.FileParser, error) {
+func (s *filesConnector) getParser(ctx context.Context, logger *slog.Logger, repo repository.Repository) (*resources.Parser, error) {
 	ns := repo.Config().Namespace
 	client, kinds, err := s.client.New(ns) // As system user
 	if err != nil {
 		return nil, err
 	}
-	return resources.NewParser(repo, client, kinds), nil
+
+	parser := resources.NewParser(repo.Config(), client, kinds)
+	if repo.Config().Spec.Linting {
+		linterFactory := lint.NewDashboardLinterFactory()
+		cfg, err := repo.Read(ctx, logger, linterFactory.ConfigPath(), repo.Config().Spec.GitHub.Branch)
```

Comment:
how do we know the configPath?

Reply:
for now it's just the default value `.dashboard-lint` at the root. We could add the option to specify the file in the repo but that feels like an improvement. 

The goal of this PR was to have parity between the linting results in Grafana and the ones displayed in the PRs.